components/ftx/browser/ftx_service.h

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_FTX_BROWSER_FTX_SERVICE_H_
+#define BRAVE_COMPONENTS_FTX_BROWSER_FTX_SERVICE_H_
+
+#include <list>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/callback_forward.h"
+#include "base/containers/queue.h"
+#include "base/files/file_path.h"
+#include "base/macros.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/scoped_observer.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "url/gurl.h"
+
+namespace base {
+class FilePath;
+class SequencedTaskRunner;
+}  // namespace base
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+class Profile;
+
+const char get_futures_data_path[] = "/api/futures";
+const char get_market_data_path[] = "/api/markets";
+const char oauth_path[] = "/oauth";
+const char oauth_token_path[] = "/api/oauth/token";
+const char oauth_balances_path[] = "/api/wallet/balances";
+const char oauth_quote_path[] = "/api/otc/quotes";
+const char futures_filter[] = "perpetual";
+
+struct TokenPriceData {
+  std::string symbol;
+  double price;
+  double percentChangeDay;
+  double volumeDay;
+};
+
+typedef std::vector<std::map<std::string, double>> FTXChartData;
+typedef std::vector<TokenPriceData> FTXFuturesData;
+typedef std::map<std::string, double> FTXAccountBalances;
+
+class FTXService : public KeyedService {
```

Comment:
can we have some comments explaining what this is since FTX is not exactly intuitive? :)

Reply:
fixed via https://github.com/brave/brave-core/compare/6f4c503619e82a440ef7db9782288e922c37d5e4..f4d887c0baf39543331107f2e22f9b0ad4986c02